ocrd_anybaseocr/cli/ocrd_anybaseocr_dewarp.py

```diff
@@ -1,184 +1,185 @@
-import torch
+# pylint: disable=wrong-import-order, import-error, too-few-public-methods
+# pylint: disable=too-many-locals, line-too-long, invalid-name, too-many-arguments
+# pylint: disable=missing-docstring
+
 import sys
 import os
-import shutil
 
 from ..constants import OCRD_TOOL
 
 from ocrd import Processor
+from ocrd_models.ocrd_page import (
+    to_xml,
+    AlternativeImageType,
+    MetadataItemType,
+    LabelsType, LabelType)
 
 from ocrd_utils import getLogger, concat_padded, MIMETYPE_PAGE
 from ocrd_modelfactory import page_from_file
 from pylab import array
 from pathlib import Path
-from PIL import Image
+
+import torch
 import ocrolib
-from ocrd_models.ocrd_page import (
-    to_xml, 
-    AlternativeImageType,
-    MetadataItemType,
-    LabelsType, LabelType
-    )
+
+from ..pix2pixhd.options.test_options import TestOptions
+from ..pix2pixhd.models.models import create_model
+from ..pix2pixhd.data.data_loader import CreateDataLoader
 
 TOOL = 'ocrd-anybaseocr-dewarp'
 LOG = getLogger('OcrdAnybaseocrDewarper')
-FALLBACK_IMAGE_GRP = 'OCR-D-IMG-DEWARP'
+
+def prepare_data(opt, page_img):
+
+    data_loader = CreateDataLoader(opt)
+    data_loader.dataset.A_paths = [page_img.filename]
+    data_loader.dataset.dataset_size = len(data_loader.dataset.A_paths)
+    data_loader.dataloader = torch.utils.data.DataLoader(data_loader.dataset,
+                                                         batch_size=opt.batchSize,
+                                                         shuffle=not opt.serial_batches,
+                                                         num_workers=int(opt.nThreads))
+    dataset = data_loader.load_data()
+    return dataset
+
+def prepare_options(gpu_id, dataroot, model_path, resize_or_crop, loadSize, fineSize):
+    # TODO oh boy
+    sys.argv = ['python']
+    sys.argv.extend(['--gpu_ids', str(gpu_id)])
+    sys.argv.extend(['--nThreads', str(1)])   # test code only supports nThreads = 1
+    sys.argv.extend(['--batchSize', str(1)])  # test code only supports batchSize = 1
+    sys.argv.extend(['--serial_batches'])  # no shuffle
+    sys.argv.extend(['--no_flip'])  # no flip
+    sys.argv.extend(['--dataroot', dataroot])
+    sys.argv.extend(['--checkpoints_dir', str(model_path.parents[1])])
+    sys.argv.extend(['--name', model_path.parents[0].name])
+    sys.argv.extend(['--label_nc', str(0)])
+    sys.argv.extend(['--no_instance'])
+    sys.argv.extend(['--resize_or_crop', resize_or_crop])
+    sys.argv.extend(['--n_blocks_global', str(10)])
+    sys.argv.extend(['--n_local_enhancers', str(2)])
+    sys.argv.extend(['--loadSize', str(loadSize)])
+    sys.argv.extend(['--fineSize', str(fineSize)])
+    sys.argv.extend(['--model', 'pix2pixHD'])
+    sys.argv.extend(['--verbose'])
```

Comment:
> So what motivated you to handle all options that way – just for the sake of uniformity?

The problem was with how `BaseOptions.parse` is implemented in pix2pixHD based on `argparse`. I cannot explain why but the approach to let pix2pixHD fill the `TestOptions` instance with `argparse` default values and then modifying the instance did not work, the overrides were simply ignored. The only way I got pix2pixHD to reliably pick up the overrides was this `sys.argv` approach. It's ugly, true, but so is using `argparse` as an API. At least this way, it is uniform as you say.

Reply:
I see. Then maybe you could drop a comment one the above circumstances (and that we plan to avoid `parse` in the future, using `torch.cuda.set_device` ourselves)?

Reply:
441d48e